airflow/providers/amazon/aws/utils/connection_wrapper.py

```diff
@@ -61,7 +99,7 @@ class AwsConnectionWrapper(LoggingMixin):
         3. The wrapper's default value
     """
 
-    conn: InitVar[Connection | AwsConnectionWrapper | None]
+    conn: InitVar[Connection | AwsConnectionWrapper | _ConnectionMetadata | None]
```

Comment:
I think we should consolidate these types a bit, perhaps with a `Protocol`.

Reply:
Yeah, I agree with you. Main problem with this class that when I initially create this helper I added InitVar and add some limitation with define attributes for this class. I have a plan to refactor it later and also get rid of `_ConnectionMetadata`.